File: src/Client/UI/Position.ts
/**
 * Defines the placement of an element.
 */
export const Position = Object.freeze({

	/**
	 * Top left.
	 */
	TopLeft: "TopLeft",

	/**
	 * Top center.
	 */
	TopCenter: "TopCenter",

	/**
	 * Top right.
	 */
	TopRight: "TopRight",

	/**
	 * Middle left.
	 */
	MiddleLeft: "MiddleLeft",

	/**
	 * Middle center.
	 */
	MiddleCenter: "MiddleCenter",

	/**
	 * Middle right.
	 */
	MiddleRight: "MiddleRight",

	/**
	 * Bottom left.
	 */
	BottomLeft: "BottomLeft",

	/**
	 * Bottom center.
	 */
	BottomCenter: "BottomCenter",

	/**
	 * Bottom right.
	 */
	BottomRight: "BottomRight"
});

/**
 * Defines the placement of an element.
 */
export type Position = typeof Position[keyof typeof Position];

File: src/Server/UI/Position.cs
namespace Belin.Base.UI;

using System.Text.Json.Serialization;

/// <summary>
/// Defines the placement of an element.
/// </summary>
[JsonConverter(typeof(JsonStringEnumConverter))]
public enum Position {

	/// <summary>
	/// Top left.
	/// </summary>
	TopLeft,

	/// <summary>
	/// Top center.
	/// </summary>
	TopCenter,

	/// <summary>
	/// Top right.
	/// </summary>
	TopRight,

	/// <summary>
	/// Middle left.
	/// </summary>
	MiddleLeft,

	/// <summary>
	/// Middle center.
	/// </summary>
	MiddleCenter,

	/// <summary>
	/// Middle right.
	/// </summary>
	MiddleRight,

	/// <summary>
	/// Bottom left.
	/// </summary>
	BottomLeft,

	/// <summary>
	/// Bottom center.
	/// </summary>
	BottomCenter,

	/// <summary>
	/// Bottom right.
	/// </summary>
	BottomRight
}
